### ISSUE TITLE
Units test is broken

### ISSUE THREAD

**Issue by juusechec** [Hashnode/mern-starter#308]
The test of this is broken. I'm forced to used "mockgoose": "^7.3.1" and the unit test crashed.

```bash
  /\_/\  Hola JORGE
 ( o.o ) Bienvenid@ al Grupo GNU/Linux UD 
  > ^ < 
jorge@ulises:~/workspace> git clone https://github.com/Hashnode/mern-starter.git
jorge@ulises:~/workspace> cd mern-starter/
jorge@ulises:~/workspace> npm install
jorge@ulises:~/workspace/mern-starter> npm run test

> mern-starter@2.0.0 test /home/jorge/workspace/mern-starter
> cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node --harmony-proxies node_modules/.bin/ava

Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
Warning: RouterContext: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
MERN is running on port: 8080! Build something amazing!
Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
Warning: Failed prop type: The prop `intl` is marked as required in `PostCreateWidget`, but its value is `undefined`.
    in PostCreateWidget
Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
[React Intl] Missing message: "createNewPost" for locale: "en"
[React Intl] Cannot format message: "createNewPost", using message id as fallback.
[React Intl] Missing message: "submit" for locale: "en"
[React Intl] Cannot format message: "submit", using message id as fallback.
[React Intl] Missing message: "createNewPost" for locale: "en"
[React Intl] Cannot format message: "createNewPost", using message id as fallback.
[React Intl] Missing message: "submit" for locale: "en"
[React Intl] Cannot format message: "submit", using message id as fallback.
[React Intl] Missing message: "by" for locale: "en"
[React Intl] Cannot format message: "by", using message id as fallback.
[React Intl] Missing message: "deletePost" for locale: "en"
[React Intl] Cannot format message: "deletePost", using message id as fallback.
[React Intl] Missing message: "createNewPost" for locale: "en"
[React Intl] Cannot format message: "createNewPost", using message id as fallback.
[React Intl] Missing message: "submit" for locale: "en"
[React Intl] Cannot format message: "submit", using message id as fallback.
[React Intl] Missing message: "switchLanguage" for locale: "en"
[React Intl] Cannot format message: "switchLanguage", using message id as fallback.
[React Intl] Missing message: "siteTitle" for locale: "en"
[React Intl] Cannot format message: "siteTitle", using message id as fallback.
[React Intl] Missing message: "createNewPost" for locale: "en"
[React Intl] Cannot format message: "createNewPost", using message id as fallback.
[React Intl] Missing message: "submit" for locale: "en"
[React Intl] Cannot format message: "submit", using message id as fallback.
[React Intl] Missing message: "addPost" for locale: "en"
[React Intl] Cannot format message: "addPost", using message id as fallback.
[React Intl] Missing message: "twitterMessage" for locale: "en"
[React Intl] Cannot format message: "twitterMessage", using message id as fallback.

   33 passed
   2 exceptions


   1. Uncaught Exception
   TypeError: bin_path is not a function
    install (/home/jorge/workspace/mern-starter/node_modules/mongodb-prebuilt/index.js:158:20)
    Object.start_server (/home/jorge/workspace/mern-starter/node_modules/mongodb-prebuilt/index.js:60:16)
    /home/jorge/workspace/mern-starter/node_modules/mockgoose/Mockgoose.js:144:42
    next (/home/jorge/workspace/mern-starter/node_modules/rimraf/rimraf.js:74:7)
    CB (/home/jorge/workspace/mern-starter/node_modules/rimraf/rimraf.js:110:9)
    FSReqWrap.oncomplete (/home/jorge/workspace/mern-starter/node_modules/rimraf/rimraf.js:229:7)


   ✖ Test results were not received from server/models/__tests__/post.spec.js
--------------------------------------------------|----------|----------|----------|----------|----------------|
File                                              |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------------------------------------------|----------|----------|----------|----------|----------------|
 client/modules/App/                              |      100 |    83.33 |       75 |      100 |                |
  App.js                                          |      100 |    66.67 |       50 |      100 |                |
  AppActions.js                                   |      100 |      100 |      100 |      100 |                |
  AppReducer.js                                   |      100 |      100 |      100 |      100 |                |
 client/modules/App/components/                   |      100 |      100 |      100 |      100 |                |
  DevTools.js                                     |      100 |      100 |      100 |      100 |                |
 client/modules/App/components/Footer/            |      100 |      100 |      100 |      100 |                |
  Footer.js                                       |      100 |      100 |      100 |      100 |                |
 client/modules/App/components/Header/            |      100 |      100 |      100 |      100 |                |
  Header.js                                       |      100 |      100 |      100 |      100 |                |
 client/modules/Intl/                             |      100 |    90.91 |       50 |      100 |                |
  IntlActions.js                                  |      100 |      100 |      100 |      100 |                |
  IntlReducer.js                                  |      100 |    90.91 |      100 |      100 |                |
  IntlWrapper.js                                  |      100 |      100 |        0 |      100 |                |
 client/modules/Post/                             |      100 |      100 |    30.77 |      100 |                |
  PostActions.js                                  |      100 |      100 |       25 |      100 |                |
  PostReducer.js                                  |      100 |      100 |      100 |      100 |                |
 client/modules/Post/components/                  |      100 |      100 |      100 |      100 |                |
  PostList.js                                     |      100 |      100 |      100 |      100 |                |
 client/modules/Post/components/PostCreateWidget/ |      100 |      100 |      100 |      100 |                |
  PostCreateWidget.js                             |      100 |      100 |      100 |      100 |                |
 client/modules/Post/components/PostListItem/     |      100 |      100 |      100 |      100 |                |
  PostListItem.js                                 |      100 |      100 |      100 |      100 |                |
 client/modules/Post/pages/PostDetailPage/        |      100 |      100 |        0 |      100 |                |
  PostDetailPage.js                               |      100 |      100 |        0 |      100 |                |
 client/modules/Post/pages/PostListPage/          |       50 |        0 |        0 |       50 |                |
  PostListPage.js                                 |       50 |        0 |        0 |       50 |             22 |
 client/util/                                     |      100 |    85.71 |      100 |      100 |                |
  apiCaller.js                                    |      100 |    85.71 |      100 |      100 |                |
  react-intl-test-helper.js                       |      100 |      100 |      100 |      100 |                |
 server/                                          |       50 |    11.54 |    28.57 |       50 |                |
  server.js                                       |       50 |    11.54 |    28.57 |       50 | 45,108,112,116 |
 server/controllers/                              |    55.56 |     7.69 |    22.22 |    55.56 |                |
  post.controller.js                              |    55.56 |     7.69 |    22.22 |    55.56 |    28,42,57,72 |
 server/models/                                   |      100 |      100 |      100 |      100 |                |
  post.js                                         |      100 |      100 |      100 |      100 |                |
 server/routes/                                   |      100 |      100 |      100 |      100 |                |
  post.routes.js                                  |      100 |      100 |      100 |      100 |                |
 server/util/                                     |       50 |        0 |        0 |       50 |                |
  fetchData.js                                    |      100 |        0 |        0 |      100 |                |
  promiseUtils.js                                 |       50 |        0 |        0 |       50 |              9 |
--------------------------------------------------|----------|----------|----------|----------|----------------|
All files                                         |    77.78 |    50.94 |    39.29 |    77.78 |                |
--------------------------------------------------|----------|----------|----------|----------|----------------|


npm ERR! Linux 4.4.62-18.6-default
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "test"
npm ERR! node v6.10.3
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! mern-starter@2.0.0 test: `cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node --harmony-proxies node_modules/.bin/ava`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mern-starter@2.0.0 test script 'cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node --harmony-proxies node_modules/.bin/ava'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mern-starter package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     cross-env NODE_ENV=test PORT=8080 MONGO_URL=mongodb://localhost:27017/mern-test node_modules/.bin/nyc node --harmony-proxies node_modules/.bin/ava
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mern-starter
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mern-starter
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/jorge/workspace/mern-starter/npm-debug.log

```

I'm using npm 3.10.10 and node v6.10.3 installed via "n" https://github.com/tj/n.

Related to https://github.com/Mockgoose/Mockgoose/issues/28

**Comment by theUtherSide** [comment]
I'm seeing this too with node 8.9.1.  Also tried node 7.7.4.  Both with npm 5.5.1

**Comment by felipe-pereira** [comment]
Tests failing here too: node.exe bad option: --harmony-proxies
node v8.9.4
npm  v5.6.0

**Comment by EternallLight** [comment]
The same here.
Fixed so far by removing the --hardmony-proxies from the npm test task and upgrading mongoose to 4.13.12. Not sure what consequences it may cause but so far tests successfully run till the end.

**Comment by mannyhenri** [comment]
@MrChristofferson up to take a look at the test issues? This issue might be related to #250

**Comment by mattlubner** [comment]
This may be fixed by #367, could you test that branch and post results here?
